schemas/observer.yml

```diff
@@ -31,6 +31,18 @@
       type: keyword
       description: >
         Hostname of the observer.
+    - name: name
+      level: core
+      type: keyword
+      description: >
+        Name of the observer.
```

Comment:
Please flesh out this description a little more, to make it clear that this is meant to be user-configured.

For inspiration, here's two other places in ECS where we have `.name` meant to be a user-customizable value: [agent.name](https://github.com/elastic/ecs/blob/v1.1.0/schemas/agent.yml#L30-L38) and [host.name](https://github.com/elastic/ecs/blob/v1.1.0/schemas/host.yml#L29-L33)